### PR TITLE
[Components.Engine.Transform] remove shadow variable in ROIValueMapper

### PR DESCRIPTION
**Pull request by maddin200**
______________________________________________________

By submitting this pull request, I acknowledge that  
**I have read, understand, and agree [SOFA Developer Certificate of Origin (DCO)](https://github.com/sofa-framework/sofa/blob/master/CONTRIBUTING.md#sofa-developer-certificate-of-origin-dco)**.
______________________________________________________

**Reviewers will merge this pull-request only if**  
- it builds with SUCCESS for all platforms on the CI.
- it does not generate new warnings.
- it does not generate new unit test failures.
- it does not generate new scene test failures.
- it does not break API compatibility.
- it is more than 1 week old (or has fast-merge label).

### CODE REVIEW

**Comment by alxbilger** [comment]
[ci-build][with-all-tests]

**Review by epernod — APPROVED** [review]

File: Sofa/Component/Engine/Transform/src/sofa/component/engine/transform/ROIValueMapper.h [hunk 1]
```diff
@@ -126,7 +126,7 @@ class ROIValueMapper : public sofa::core::DataEngine
                 {
                     size_t oldSize = outputValues.size();
                     outputValues.resize(ind+1);
-                    for (size_t j=oldSize ; j<ind+1 ; j++) outputValues[j] = defaultValue;
+                    for (size_t kk=oldSize ; kk <ind+1 ; kk++) outputValues[kk] = defaultValue;
```
Comment (epernod) [inline Sofa/Component/Engine/Transform/src/sofa/component/engine/transform/ROIValueMapper.h]:
```suggestion
                    for (size_t kk=oldSize ; kk <ind+1 ; kk++) 
                    {
                        outputValues[kk] = defaultValue;
                    }
```

**Review by epernod — CHANGES_REQUESTED** [review]

**Comment by damienmarchal** [comment]
@epernod  is the requested change done in this one ?

**Comment by alxbilger** [comment]
I made the code more concise, more efficient and clearer to me. It also removes the need of the variable which was initially shadowed. Let me know if you agree/disagree.
I also merged master.
I suggest also to move the methods definitions in the cpp file, but I did not want to be out of the scope of this PR.

**Comment by maddin200** [comment]
Hi thanks for help - any update is fine go ahead :-)

**Review by epernod — APPROVED** [review]

**Pull request merged**